### PR TITLE
refactor(typescript): landing

### PR DESCRIPTION
**Pull request by olros**
## Description

#243 and #255 

Comments/issues/screenshots:
- Refactored the Landing-page and its components to Typescript
- Change name of landing-folder from "NewLanding" to "Landing"
- Add global overrides for link color to remove the green links, uses the primary color instead
- Update wave with svg's instead of png's to render the waves. This allows to color to be dynamic and are set to equal the background.
- No big visual changes, the waves may look a little bit different

image

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] The PR includes a `closes #issueID`
- [x] The PR includes a picture showing visual changes

## Typescript convert checklist

- [x] All relative imports has been converted to absolute imports
- [x] All HOC's have been replaced with Hooks
- [x] Uses theme.spacing() instead of custom padding
- [x] Uses theme.breakpoints instead of custom media queries

### CODE REVIEW

File: src/containers/Landing/components/EventsListView.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  } else if (!events.length) {
+    return (
+      <Typography align='center' className={classes.noEventText} variant='subtitle1'>
+        Ingen kommende arrangementer
+      </Typography>
+    );
+  } else {
+    return (
+      <div className={classes.container}>
+        {events.map((event, index) => index < 3 && <ListItem event={event} key={event.id} />)}
```
Comment (Zenjjim) [inline src/containers/Landing/components/EventsListView.tsx:58]: make "3" a constant with a propper name

File: src/containers/Landing/components/EventsView.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  useEffect(() => {
+    getEvents()
+      .then((eventObject) => setEvents(eventObject.results))
+      .finally(() => setIsLoading(false));
+  }, [getEvents]);
+
+  useEffect(() => {
+    // Load expired events when switching to calendar tab and they havn't been loaded already
+    if (!oldEvents.length && tab === 1) {
```
Comment (Zenjjim) [inline src/containers/Landing/components/EventsView.tsx:42]: make 0 and 1 enums for better readability

File: src/containers/Landing/components/NewsListView.tsx [hunk 3]
```diff
@@ -1,90 +1,78 @@
...
-      <div className={classes.noEventText}>
+      <div className={classes.noNewsText}>
         <CircularProgress className={classes.progress} />
       </div>
     );
   } else if (news.length) {
     return (
       <>
-        <div className={classes.newsListContainer}>{news.map((newsItem, index) => index < 2 && <ListItem key={index} news={newsItem} />)}</div>
+        <div className={classes.container}>{news.map((newsItem, index) => index < 2 && <ListItem key={index} news={newsItem} />)}</div>
```
Comment (Zenjjim) [inline src/containers/Landing/components/NewsListView.tsx:63]: same as above. make 2 a constant with a name

File: src/containers/Landing/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import Helmet from 'react-helmet';
```
Comment (Zenjjim) [inline src/containers/Landing/index.tsx:2]: is helmet even used?

File: src/containers/Landing/index.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}));
+
+const Landing = () => {
+  const classes = useStyles();
+  return (
+    <Navigation banner={<Wave />} fancyNavbar maxWidth={false}>
+      <Helmet>
+        <title>Forsiden - TIHLDE</title>
+      </Helmet>
+      <div className={classes.smoke}>
```
Comment (Zenjjim) [inline src/containers/Landing/index.tsx:37]: would not a smoke component be good since its used almost everywhere?

File: src/index.tsx [hunk 6]
```diff
@@ -89,6 +90,7 @@ export const Providers = ({ children }: ProvidersProps) => {
           <JobPostProvider>
```
Comment (Zenjjim) [inline src/index.tsx:90]: not a part of this PR but index.tsx is getting big. Sould splitt ut ProviderHandler and Router to seprate files

**Review by Zenjjim — CHANGES_REQUESTED** [review]
Good joob. A couple of comments about clean code but logic looks good.

**Comment by Zenjjim** [comment]
image
There is a line that should get fixed right below the waves

File: src/index.tsx (hunk 6 above)
Comment (olros) [reply src/index.tsx:90]: I agree

File: src/assets/css/index.css [hunk 7]
```diff
@@ -3,19 +3,11 @@
...
 }
 body {
     margin: 0;
     font-family: 'Roboto', sans-serif;
 }
-
-/* LINKS */
 a {
     color: #009a3e;
 }
```
[the remark this answers is not in the document]
Comment (olros) [reply src/assets/css/index.css:13]: Thought I'd leave it as a "backup", but it may not be necessary

File: src/containers/Landing/index.tsx (hunk 4 above)
Comment (olros) [reply src/containers/Landing/index.tsx:2]: It works quite fine actually:
image

File: src/containers/Landing/index.tsx (hunk 5 above)
Comment (olros) [reply src/containers/Landing/index.tsx:37]: It's used like this 3 places: Landing, About and Companies. I don't think that is enough to create a new component for it

**Review by Zenjjim — APPROVED** [review]

**Pull request merged**